added interest rate feature will increase the amount of people willin…

DomBen3 opened the pull request:
…g to invest in Testudo Bank

src/test/tests/MvcControllerIntegTest.java
src/main/java/MvcController.java
Added functionality to deposit form which is incrementing the number of deposits for interest if the deposit was above 20 dollars. This makes it so that most of the functionality of the new interest rate will be in the applyInterest form.
Most of the logic for how much to deposit and if the bank should deposit for the interest is in the applyInterest form. This change make users more inclined to use the Testudo Bank services since they get more money every fifth deposit above 20 dollars.

s1ddh4rthc commented:
Hey Dominic, nice start to Assignment #2. Make sure you format your PR message better - you can search up how to format your PR messages to make them look more organized. In terms of your logic, I think it looks good. Perhaps you can combine your if statements that are checking whether to return "welcome" or not. That will help your code look cleaner. Also make sure that you're incrementing the `numDepositsForInterest` variable if the deposit amount is greater than $20.

DomBen3 commented:
**Changes Made in this PR:**

I moved the location of the apply interest function inside the deposit mapping. It is now inside the if statement that checks for a deposit of larger than 20. I also added a new test that shows the change helps with an edge case.


`src/main/java/net/testudobank/MvcController.java`
- Moved the `applyInterest()` function to be inside of the `if` statement that checks for a deposit of 20 or more.
- Before, deposits of less than 20 wouldn't add to the counter, but would still apply interest
 

`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`
- Added a new test to check for the edge case fixed in the previously stated file.
- Added a check for having a count of 5 deposits and then depositing an amount less than 20.
